### PR TITLE
New Adapter: InteractiveOffers

### PR DESCRIPTION
**Pull request by IOTiagoFaria**
md file pr : https://github.com/prebid/prebid.github.io/pull/2879/

### CODE REVIEW

File: static/bidder-info/interactiveOffers.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+maintainer:
+  email: "faria@interactiveoffers.com"
+gvlVendorID: 0
```
Comment (SyntaxNode) [inline static/bidder-info/interactiveOffers.yaml:3]: I recommend removing the `gvlVendorID` line entirely if you aren't registered with the IAB Vendor List.

File: static/bidder-params/interactiveOffers.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "title": "Interactive Offers Adapter Params",
+  "description": "A schema which validates params accepted by Interactive Offers adapter",
+  "type": "object",
+  "properties": {
+    "pubid": {
+      "type": "integer",
+      "description": "The publisher id"
+    },
+    "tmax": {
```
Comment (SyntaxNode) [inline static/bidder-params/interactiveOffers.json:11]: There is already a TMax value on the OpenRTB request. Why does your adapter need its own?

File: static/bidder-params/interactiveOffers.json [hunk 3]
```diff
@@ -0,0 +1,17 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "title": "Interactive Offers Adapter Params",
+  "description": "A schema which validates params accepted by Interactive Offers adapter",
+  "type": "object",
+  "properties": {
+    "pubid": {
```
Comment (SyntaxNode) [inline static/bidder-params/interactiveOffers.json:7]: The publisher id is already available in the OpenRTB request as either app.publisher.id or site.publisher.id. Why does your adapter need its own?

File: openrtb_ext/imp_interactiveOffers.go [hunk 4]
```diff
@@ -0,0 +1,6 @@
+package openrtb_ext
+
+type ExtImpInteractiveOffers struct {
+	Pubid int `json:"pubid"`
```
Comment (SyntaxNode) [inline openrtb_ext/imp_interactiveOffers.go:4]: Nitpick: `PubID` is preferred for the variable name by Go naming conventions.

File: adapters/interactiveOffers/interactiveOffers.go [hunk 5]
```diff
@@ -0,0 +1,79 @@
+package interactiveOffers
```
Comment (SyntaxNode) [inline adapters/interactiveOffers/interactiveOffers.go:1]: Package names in Go are preferred to be lower case. Please consider naming your package and folder `interactiveoffers`.

File: openrtb_ext/bidders.go [hunk 6]
```diff
@@ -73,113 +73,114 @@ func IsBidderNameReserved(name string) bool {
...
+	BidderEngageBDR         BidderName = "engagebdr"
+	BidderEPlanning         BidderName = "eplanning"
+	BidderEpom              BidderName = "epom"
+	BidderGamma             BidderName = "gamma"
+	BidderGamoshi           BidderName = "gamoshi"
+	BidderGrid              BidderName = "grid"
+	BidderGumGum            BidderName = "gumgum"
+	BidderImprovedigital    BidderName = "improvedigital"
+	BidderInMobi            BidderName = "inmobi"
+	BidderInteractiveOffers BidderName = "interactiveOffers"
```
Comment (SyntaxNode) [inline openrtb_ext/bidders.go:126]: IMHO it's simpler for the bidder name to be all lower case. Please consider:
```
BidderInteractiveOffers BidderName = "interactiveoffers"
```
Thoughts @bsardo?

File: adapters/interactiveOffers/params_test.go [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+		}
+	}
+}
+
+var validParams = []string{
+	`{"pubid":35}`,
+}
+
+var invalidParams = []string{
+	`{"pubid":"35"}`,
```
Comment (SyntaxNode) [inline adapters/interactiveOffers/params_test.go:40]: I like that you have a test case for the incorrect type. I also recommend tests where the required `pubid` is missing and also combinations involving `tmax`.

File: config/config.go [hunk 8]
```diff
@@ -857,6 +857,7 @@ func SetupViper(v *viper.Viper, filename string) {
 	v.SetDefault("adapters.gumgum.endpoint", "https://g2.gumgum.com/providers/prbds2s/bid")
 	v.SetDefault("adapters.improvedigital.endpoint", "http://ad.360yield.com/pbs")
 	v.SetDefault("adapters.inmobi.endpoint", "https://api.w.inmobi.com/showad/openrtb/bidder/prebid")
+	v.SetDefault("adapters.interactiveOffers.endpoint", "https://rtb.ioadx.com/bidRequest/?partnerId=d9e56d418c4825d466ee96c7a31bf1da6b62fa04")
```
Comment (SyntaxNode) [inline config/config.go:860]: I'd like to verify the partner id refers to the Prebid ecosystem and not a specific host, such as Xandr/AppNexus.

File: static/bidder-info/interactiveOffers.yaml (hunk 1 above)
Comment (IOTiagoFaria) [reply static/bidder-info/interactiveOffers.yaml]: gvlVendorId was removed

File: static/bidder-params/interactiveOffers.json (hunk 2 above)
Comment (IOTiagoFaria) [reply static/bidder-params/interactiveOffers.json]: TMax was removed, the md file was updated

File: static/bidder-params/interactiveOffers.json (hunk 3 above)
Comment (IOTiagoFaria) [reply static/bidder-params/interactiveOffers.json]: the publisher.id under the OpenRTB standards is not a required value, for us it is

File: adapters/interactiveoffers/interactiveoffers.go [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+		Method: "POST",
+		Uri:    a.endpoint,
+		Body:   requestJSON,
+	}
+
+	return []*adapters.RequestData{requestData}, nil
+}
+
+func (a *adapter) MakeBids(request *openrtb2.BidRequest, requestData *adapters.RequestData, responseData *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	if responseData.StatusCode == http.StatusNoContent {
```
Comment (SyntaxNode) [inline adapters/interactiveoffers/interactiveoffers.go:35]: Please add a json test for a no bid response of 204.

**Review by SyntaxNode — COMMENTED** [review]
Looks good. Thank you for the updates. Could you please verify the partnerId in the endpoint is not specific to a host, such as Xandr/AppNexus?

File: config/config.go (hunk 8 above)
Comment (IOTiagoFaria) [reply config/config.go]: Correct, that partner id is regarding the Prebid ecosystem

File: static/bidder-info/interactiveoffers.yaml [hunk 10]
```diff
@@ -0,0 +1,9 @@
+maintainer:
+  email: "faria@interactiveoffers.com"
```
[the remark this answers is not in the document]
Comment (SyntaxNode) [reply static/bidder-info/interactiveoffers.yaml:2]: Good point. Please use a group email address.

**Comment by IOTiagoFaria** [comment]
Hi @bsardo , what's the status on this review? do I need to do something?

**Comment by bsardo** [comment]
Hi @IOTiagoFaria, please see the two comments above from me and @SyntaxNode. We were just waiting on you to make two changes:
- add a json test for a no bid response of 204
- change the email address in the yaml to a group email

Everything else looks good from my perspective.

**Comment by IOTiagoFaria** [comment]
I really though that I had committed those changes before

**Comment by SyntaxNode** [comment]
The checks are failing due to a code formatting issue. Please run gofmt to apply standard formatting to your changes.

**Comment by bsardo** [comment]
It looks like something happened when you merged with master because `BidderAdf` is missing in the slice returned by `CoreBidderNames` in `openrtb_ext/bidders.go`. That was merged in last week.

**Comment by IOTiagoFaria** [comment]
Sorry guys, but I'll just restart this process, I'll close this PR and create a new one without any issues

**Pull request closed without merging**